library/src/loader_impl.cxx

```diff
@@ -168,202 +148,145 @@ class loader_impl::internals
     this->Window.setAnimationNameInfo(this->AnimationManager.GetAnimationName());
 
-    // Display the importer description
-    loader_impl::internals::DisplayImporterDescription(this->GenericImporter);
-
-    // Set the importer to use for coloring and actors
-    this->Window.SetImporterForColoring(this->GenericImporter);
+    // Display output description
+    loader_impl::internals::DisplayImporterDescription(this->MetaImporter);
 
-    // Initialize renderer and reset camera to bounds
+    // Update all window options and reset camera to bounds if needed
     this->Window.UpdateDynamicOptions();
-    this->Window.getCamera().resetToBounds();
+    if (!this->Options.scene.camera.index.has_value())
+    {
+      this->Window.getCamera().resetToBounds();
+    }
 
     // Print info about scene and coloring
     this->Window.PrintColoringDescription(log::VerboseLevel::DEBUG);
     this->Window.PrintSceneDescription(log::VerboseLevel::DEBUG);
+  }
 
-    this->DefaultScene = true;
+  static void DisplayImporterDescription(vtkImporter* importer)
+  {
+    vtkIdType availCameras = importer->GetNumberOfCameras();
+    if (availCameras <= 0)
+    {
+      log::debug("No camera available");
+    }
+    else
+    {
+      log::debug("Camera(s) available are:");
+    }
+    for (int i = 0; i < availCameras; i++)
+    {
+      log::debug(i, ": ", importer->GetCameraName(i));
+    }
+    log::debug("");
+    log::debug(importer->GetOutputsDescription(), "\n");
   }
 
-  bool DefaultScene = false;
   const options& Options;
   window_impl& Window;
   interactor_impl* Interactor = nullptr;
   animationManager AnimationManager;
 
-  vtkSmartPointer<vtkImporter> CurrentFullSceneImporter;
-  vtkNew<vtkF3DGenericImporter> GenericImporter;
+  vtkNew<vtkF3DMetaImporter> MetaImporter;
 };
 
 //----------------------------------------------------------------------------
 loader_impl::loader_impl(const options& options, window_impl& window)
   : Internals(std::make_unique<loader_impl::internals>(options, window))
 {
-  // Set render window on generic importer
-  this->Internals->GenericImporter->SetRenderWindow(this->Internals->Window.GetRenderWindow());
 }
 
 //----------------------------------------------------------------------------
 loader_impl::~loader_impl() = default;
 
 //----------------------------------------------------------------------------
-loader& loader_impl::loadGeometry(const std::string& filePath, bool reset)
+loader& loader_impl::add(const fs::path& filePath)
 {
-  // Check file validity
-  if (filePath.empty())
-  {
-    // Calling with reset is an usual codepath to reset the window
-    if (!reset)
-    {
-      log::debug("Provided geometry file path is empty\n");
-    }
-    this->Internals->Reset();
-    return *this;
-  }
-  if (!vtksys::SystemTools::FileExists(filePath, true))
-  {
-    throw loader::load_failure_exception(filePath + " does not exists");
-  }
+  std::vector<fs::path> paths = { filePath };
+  return this->add(paths);
```

Comment:
Instead of creating proxy lists to forward the logic to the `add(vec<fs::path>)` function, can you do the opposite and open the file in `add(fs::path)`?

Reply:
Ok fair enough, it's not really important anyway.